### PR TITLE
Infrastructure for implementing a Kubewarden controlled vocabulary.

### PR DESCRIPTION
**Pull request by jhkrug**
docusaurus.config.js defines a preprocessor which replaces all instances of the variables defined in `./variables.json` with their values. See the changes in quick-start.md for an example. `[[< policy-server-component >]]' to 'PolicyServer'.

Part of #394.

### CODE REVIEW

**Review by fabriziosestito — APPROVED** [review]

File: docs/quick-start.md
Comment (viccuad) [inline docs/quick-start.md:21]:
```suggestion
* Some `[[< policy-server-component >]]` resources: representing a deployment of a Kubewarden policy-server. Your administrator's policies are loaded and evaluated by the Kubewarden `[[< policy-server-component >]]`
```

Maybe?

As in, the Custom Resource PolicyServer represents deployment of the policy-server image/binary.

File: docs/quick-start.md
Comment (viccuad) [inline docs/quick-start.md:100]: @kubewarden/kubewarden-developers  can we drop this caution block already?

File: docs/quick-start.md
Comment (viccuad) [inline docs/quick-start.md:133]:
```suggestion
kind: PolicyServer
```

We shouldn't do it inside code blocks, as they are copy-pasted when trying examples.

File: variables.json
Comment (viccuad) [inline variables.json:3]: Quibble: Maybe better to name the variable `policy-server-customresource`, or `policy-server-cr` (written PolicyServer), in preparation for having a `policy-server-image`, or `policy-server-pod`, or `policy-server-binary` (written policy-server).

**Review by viccuad — COMMENTED** [review]
Added some review comments.

Could the policy-server-component variable for "PolicyServer" link to a glossary page with a paragraph explanation?
I would like to picture how the different terms can be listed.

**Comment by jhkrug** [comment]
> Added some review comments.
> 
> Could the policy-server-component variable for "PolicyServer" link to a glossary page with a paragraph explanation? I would like to picture how the different terms can be listed.

Yes, that could be done.

File: variables.json
Comment (jhkrug) [reply variables.json:3]: It's just an example to show how it works, constructing the list of variable_names and variable_values will be quite an intensive joint effort. I think I'll move this to draft, as a reminder to remove the changes to quick-start.md before merging. They come after, the name and values are decided.

File: docs/quick-start.md
Comment (jhkrug) [reply docs/quick-start.md:21]: I guess this is where we will, in due course, have [[< policy-server-deployment >]] mapping to policy-server.

File: docs/quick-start.md
Comment (jhkrug) [reply docs/quick-start.md:100]: Just using the quick start for examples of how this might work for now.

**Review by jvanz — APPROVED** [review]

File: variables.json
Comment (viccuad) [reply variables.json:3]: fair enough, no need to block on this for this pr. Resolving.

File: docs/quick-start.md
Comment (viccuad) [reply docs/quick-start.md:133]: Oh, yep, they would not copy-paste from the markdown source.
I was on the impression that the substitution would be for a "PolicyServer" that is a link to a reference page or something. If that's the case, then the substition shouldn't be part of the yaml block I suppose. If not, disregard.

**Comment by viccuad** [comment]
> Also, we may want to think about TermWeb (I think you can get to it now, via Okta B2C), SUSEs new terminology database, as opposed to a in-house glossary.

I would prefer to link to a local glossary part of our docs. We are part of CNCF yaddayadda, plus ideally we should ship the docs somehow, in an airgap installation, without needing to hit external URLs.

File: docs/quick-start.md
Comment (jhkrug) [reply docs/quick-start.md:133]: Yeah, might be getting a little more complex. Would need variables policy-server-component and policy-server-component-in-a-code-block, or the variable replacement code needs to not add a link to a glossary if inside a code block. Hmmm.

**Comment by jhkrug** [comment]
A discussion. Considering, for the moment, just the term PolicyServer, capitalized in the Kubernetes style, it is a Kubewarden 'component'.

So, how should we be seeing PolicyServer in the documentation? It appears in both inline code-blocks and fenced code-blocks. When it appears in the text, in a sentence, is it always in an inline code-block? If not, should it be enclosed in either single or double quotation marks? Are we trying to indicate a difference between a PolicyServer in the text, and a PolicyServer in a code-block? I hope not, and I don't think we are. 

So, if we want PolicyServer in the text to be both in an inline code-block for formatting/semantic reasons and we also want it to be a tooltip and a link to the glossary then the expansion of `[[> policy-server-component <]]` can be configured to look exactly like it currently does in an inline code-block, but colored a little differently to indicate it's a tooltip/link.

This is moving away from the semantic meaning of the markdown text which I'm reluctant to do.

Hmm, more thinking needed.

**Comment by jhkrug** [comment]
Take a look at https://deploy-preview-396--silly-bunny-8cedd0.netlify.app/next/quick-start the first paragraph of bullet points for a better idea of how it might look.

Also please take a look at the glossary. Some terms are probably unnecessary there and some need to be added. If you want to contribute some pithy text for any of the entries, cool!

**Comment by jvanz** [comment]
I think the glossary is great! We are moving in the right direction. I'm not sure where we use all the terms, but I would say we can reduce the initial terms count there just for the ones more directed related to Kubewarden.  Keeping only:

- AdmissionPolicy
- ClusterAdmissionPolicy
- ClusterPolicyReport
- kwctl
- MutatingWebhookConfiguration
- PolicyReport
- PolicyServer
- ValidatingWebhookConfiguration
- waPC
- WASI
- Wasm
- Wasmtime


Or maybe keeping the term used in the Quick Start for now. I don't know... I'm just wondering about that.

File: docs/glossary.md
Comment (jvanz) [inline docs/glossary.md:39]: The tools also allows the user to generate the Kubernetes yaml files used to deploy the policies.

File: docs/glossary.md
Comment (jvanz) [inline docs/glossary.md:27]: This is the same of the `AdmissionPolicy`. But this targets cluster-wide resources.

File: docs/glossary.md
Comment (jvanz) [inline docs/glossary.md:43]: This is the [Kubernetes resource](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) created by the Kubewarden controller to let Kubernetes knows where it should sent the `AdmissionReview`. In other words, this is how Kubewarden controller teach Kubernetes where to find the policy which could mutate resources before deploy them in the cluster

File: docs/glossary.md
Comment (jvanz) [inline docs/glossary.md:58]: This is the [Kubernetes resource](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) created by the Kubewarden controller to let Kubernetes knows where it should sent the `AdmissionReview`. In other words, this is how Kubewarden controller teach Kubernetes where to find the policy which  validates resources before deploy them in the cluster

**Review by jvanz — CHANGES_REQUESTED** [review]

File: docs/glossary.md
Comment (jhkrug) [reply docs/glossary.md:58]: Thanks.

**Comment by jhkrug** [comment]
Ready for review again. Keeping it in draft, I'm still not totally convinced. See what you think. The quick start page is the only one modified (so far) to use this idea of the controlled vocabulary.

**Review by jvanz — COMMENTED** [review]
LGTM

The glossary items and the position of the tooltips looks fine to me. I'm just not sure about the colors. But I do not know how to improve it.

**Comment by jhkrug** [comment]
> LGTM
> 
> The glossary items and the position of the tooltips looks fine to me. I'm just not sure about the colors. But I do not know how to improve it.

Yes, it makes the page quite busy. I'm tempted to try no color in the text, so no visual indication at all that there is a tooltip. So, once you know there may be one, you can use it if you want to. It's not quite as invisible as it seems. It's surprising how much we use a mouse/pointer when reading.

**Comment by jhkrug** [comment]
Ok, I think this is the final version. Much more subtle.

**Comment by jvanz** [comment]
> Ok, I think this is the final version. Much more subtle.

That's it! I like it

**Review by jvanz — APPROVED** [review]

**Comment by jhkrug** [comment]
I think this can be merged. It could be improved further, by adding some code to reduce duplication in variables.json. For a future iteration, when docs infrastructure is decided on.

**Pull request merged**
